src/python/pants/init/engine_initializer.py

```diff
@@ -371,6 +374,19 @@ def union_membership_singleton() -> UnionMembership:
     def build_root_singleton() -> BuildRoot:
       return BuildRoot.instance
 
+    @rule
+    def single_build_file_address(specs: Specs) -> BuildFileAddress:
+      build_file_addresses = yield Get(BuildFileAddresses, Specs, specs)
+      if len(build_file_addresses.dependencies) != 1:
+        provenanced_addresses = yield Get(ProvenancedBuildFileAddresses, Specs, specs)
+        matched_addrs = [addr.build_file_address.to_address() for addr in provenanced_addresses]
```

Comment:
You go through the effort of grabbing `ProvenancedBuildFileAddresses` and then immidiately through away the extra information in each `ProvenancedBuildFileAddress` by reaching in for its `build_file_address` and ignoring its `provenance` (`Spec`). Did you intend to perhaps provide a roster of targets each spec expanded to in the case of multiple specs? IE, given `./pants src/python/company/tools/tool1:bin src/python/company/apps::` fail with something like?:
```
From 'src/python/company/tools/tool1:bin':
  src/python/company/tools/tool1:bin

From 'src/python/company/apps::':
  src/python/company/apps/app1:bin
  src/python/company/apps/app2:bin
  src/python/company/apps/app3:bin
```